components/brave_wallet/browser/sns_resolver_task.h

```diff
@@ -0,0 +1,213 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_SNS_RESOLVER_TASK_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_SNS_RESOLVER_TASK_H_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/memory/weak_ptr.h"
+#include "base/notreached.h"
+#include "brave/components/api_request_helper/api_request_helper.h"
+#include "brave/components/brave_wallet/browser/solana_keyring.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "brave/components/brave_wallet/common/solana_address.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_wallet {
+
+using SnsNamehash = std::array<uint8_t, 32>;
+
+SnsNamehash GetHashedName(const std::string& name);
+
+absl::optional<SolanaAddress> GetNameAccountKey(
+    const SnsNamehash& hashed_name,
+    const absl::optional<SolanaAddress>& parent);
+
+absl::optional<SolanaAddress> GetMintAddress(
+    const SolanaAddress& domain_address);
+absl::optional<SolanaAddress> GetDomainKey(const std::string& domain,
+                                           bool record);
```

Comment:
Unit tests for these?

Reply:
https://github.com/brave/brave-core/pull/15617/commits/1e34bd9363c2e1aa4593c00f3409b8c119ccab29